packages/client/hmi-client/src/composables/useCharts.ts

```diff
@@ -541,6 +543,53 @@ export function useCharts(
 		return parameterDistributionCharts;
 	};
 
+	const useWeightsDistributionCharts = () => {
+		const WEIGHT_PARAM_NAME = 'weight_param';
+		const weightsCharts = computed(() => {
+			const charts: VisualizationSpec[] = [];
+			if (!isChartReadyToBuild.value) return charts;
+
+			// Model configs are used to get the model config metadata. This order of model configs in arrays are not guaranteed to be the same as the order of model configs in the pyciemss results
+			const modelConfigs = <ModelConfiguration[]>modelConfig?.value ?? [];
+			// extractModelConfigIdsInOrder ensures that the order of model config IDs are matched with the order of corresponding model index in the pyciemss results
+			const modelConfigIds = extractModelConfigIdsInOrder(chartData.value?.pyciemssMap ?? {});
+
+			const data = chartData.value?.result.filter((d) => d.timepoint_id === 0) ?? [];
+			const labelBefore = 'Before calibration';
+			const labelAfter = 'After calibration';
+
+			const colors = CATEGORICAL_SCHEME.slice(1); // exclude the first color which is for ensemble variable
+
+			modelConfigIds.forEach((configId, index) => {
+				const modelConfigName = getModelConfigName(modelConfigs, configId ?? '');
```

Comment:
Are you doing `configId ?? ''` because typescript is complaining? It doesn't seem possible for configId to be null/undefined at this stage, and if they are null there are much greater problems. Or am I wrong here and you can actually get null configs?

Reply:
Yeah configId is always string and we don't need `?? ''`. Probably it was a copy/paste mistake. Typescript compiler was fine.